comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -263,6 +318,10 @@ func (r *secretResolver) resolve(data []byte, origin string, notifyCb secrets.Re
 		if r.fetchHookFunc != nil {
 			// hook used only for tests
 			secrets, err = r.fetchHookFunc(newHandles)
+			// add results to the cache
+			for handle, secretValue := range secrets {
+				r.cache[handle] = secretValue
+			}
 		} else {
 			secrets, err = r.fetchSecret(newHandles)
 		}
```

Comment:
Weirdly, we have to manually handle the cache store for the test case branch. It looks like `fetchSecret` is not only fetching the secret but also updateiung the cache storage. 

I think we could improve the readbility ofthe code by having the cache by handle ins the same pace for both branches: `fetchHookFunc` , and `fetchSecret`

What about code that looks like this:
```
if r.fetchHookFunc != nil {
	// hook used only for tests
	secrets, err = r.fetchHookFunc(newHandles)
} else {
	secrets, err = r.fetchSecret(newHandles)
}

// add results to the cache
for handle, secretValue := range secrets {
	r.cache[handle] = secretValue
}

```

Reply:
For sure. During development, an earlier version of the code did this, but I realized TestFetchSecret depended upon the cache getting updated inside `r.fetchSecret`. However, I think you're right that this reads much better, and there's no need to keep that test focusing on such implementation details.